src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/ActionService.java

```diff
@@ -144,6 +144,12 @@ public void processTick(int shardNumber) {
         }
     }
 
+    @VisibleForTesting
+    boolean needToCheckSla(int hashCode, int shardNumber) {
+        // hashCode can be negative, so we can't use expression `hashCode() % shardCount == shardNumber`
+        return (hashCode + shardNumber) % shardCount == 0;
```

Comment:
Didn't get it. As far as I can see this sum still may be negative.

Reply:
Nice trick.